saleor/discount/tests/test_utils/test_refresh_order_prices_and_discounts.py

```diff
@@ -0,0 +1,1079 @@
+from decimal import Decimal
+
+import graphene
+import pytest
+
+from ....core.prices import quantize_price
+from ....core.taxes import zero_money
+from ....order import OrderStatus
+from ....product.models import Product
+from ....product.utils.variant_prices import update_discounted_prices_for_promotion
+from ....product.utils.variants import fetch_variants_for_promotion_rules
+from ... import DiscountType, DiscountValueType, RewardValueType, VoucherType
+from ...models import OrderLineDiscount, Promotion, PromotionRule
+from ...utils.order import refresh_order_base_prices_and_discounts
+from ...utils.voucher import create_or_update_voucher_discount_objects_for_order
+
+
+@pytest.fixture
+def order_with_lines(order_with_lines):
```

Comment:
You shouldn't override existing fixture. Create a new one instead. 
```suggestion
def draft_order_with_lines(order_with_lines):
````

Reply:
Creating a new fixture will not give the same result. Overwriting the existing `order_with_lines` will also change the order status of all the fixtures which inherit from `order_with_lines`. This way we save multiple database calls. What do you think?

Reply:
To avoid rebases, I change it in next PR of the project: https://github.com/saleor/saleor/pull/17419/commits/16b6d3e78ccd1531f0d6aadc07a1c0080518e2be